lib/modules/datasource/rpm/index.ts

```diff
@@ -0,0 +1,220 @@
+        { datasource: RpmDatasource.id, url: repomdUrl },
+        'Invalid response format',
+      );
+      throw new Error(
+        `${repomdUrl} is not in XML format. Response body: ${response.body}`,
+      );
+    }
+
+    // parse repomd.xml using XmlDocument
+    const xml = new XmlDocument(response.body);
+
+    const filelistsData = xml.childWithAttribute('type', 'filelists');
+
+    if (!filelistsData) {
+      logger.warn(
+        `No filelists data found in ${repomdUrl}, xml contents: ${response.body}`,
+      );
+      throw new Error(`No filelists data found in ${repomdUrl}`);
+    }
+
+    const locationElement = filelistsData.childNamed('location');
+    if (!locationElement) {
+      throw new Error(`No location element found in ${repomdUrl}`);
+    }
+    const href = locationElement.attr.href;
+    if (!href) {
+      throw new Error(`No href found in ${repomdUrl}`);
+    }
+    // replace trailing "repodata/" from registryUrl, if it exists, with a "/" because href includes "repodata/"
+    const registryUrlWithoutRepodata = registryUrl.replace(
+      /\/repodata\/?$/,
+      '/',
+    );
+    return joinUrlParts(registryUrlWithoutRepodata, href);
+  }
+
+  async getReleasesByPackageName(
+    filelistsGzipUrl: string,
+    packageName: string,
+  ): Promise<ReleaseResult | null> {
+    let response;
+    let decompressedBuffer;
+    try {
+      // filelistsXmlUrl is a .gz file, need to extract it before parsing
+      response = await this.http.getBuffer(filelistsGzipUrl);
+      if (response.body.length === 0) {
+        logger.warn(`Empty response body from getting ${filelistsGzipUrl}.`);
+        throw new Error(
+          `Empty response body from getting ${filelistsGzipUrl}.`,
+        );
+      }
+      // decompress the gzipped file
+      decompressedBuffer = await gunzipAsync(response.body);
+    } catch (err) {
+      logger.warn(
+        `Failed to fetch or decompress ${filelistsGzipUrl}: ${
+          err instanceof Error ? err.message : err
+        }`,
+      );
+      throw err;
+    }
+    const xmlString = decompressedBuffer.toString('utf8');
+    // check if filelistsXmlUrl is in XML format
+    if (!(xmlString.startsWith('<?xml') || xmlString.startsWith('\n<?xml'))) {
+      logger.warn(
+        `Decompressed ${filelistsGzipUrl} is not in XML format. Contents: ${xmlString}`,
+      );
+      throw new Error(
+        `Decompressed ${filelistsGzipUrl} is not in XML format. Contents: ${xmlString}`,
+      );
+    }
+
+    // parse filelists.xml
+    const data = new XmlDocument(xmlString);
+    const packages = data.childrenNamed('package');
+    if (!packages || packages.length === 0) {
+      logger.warn(
+        `No packages found in ${filelistsGzipUrl}, xml contents: ${xmlString}`,
+      );
+      throw new Error(`No packages found in ${filelistsGzipUrl}`);
+    }
+    const releases: Record<string, Release> = {};
+    logger.debug(`Found ${packages.length} packages in ${filelistsGzipUrl}`);
```

Comment:
```suggestion
    logger.trace(`Found ${packages.length} packages in ${filelistsGzipUrl}`);
```

you can use `loglevelremap`[^1] for remap to higher level for debugging

[^1]: https://docs.renovatebot.com/configuration-options/#loglevelremap

Reply:
updated to trace